C/c4Database.cc

```diff
@@ -153,7 +153,12 @@ static C4DatabaseConfig newToOldConfig(const C4DatabaseConfig2 &config2) {
 
 /*static*/ void C4Database::copyNamed(slice sourcePath, slice destinationName, const Config &config) {
     ensureConfigDirExists(config);
-    FilePath from(sourcePath);
+    std::string sourcePathString = sourcePath.asString();
```

Comment:
`FilePath from(sourcePath, "");`

When doing above, the FilePath will not try to split sourcePath into dir and file and the trailing '/' will be handled correctly as it is telling that sourcePath is a directory. We are doing this in multiple places in LiteCore.

Reply:
yes, this is a nice way telling the library that its a directory. The issue is whether should we rely on the syntactic input to do the recursive copy. Here is the snippet #if __APPLE__,
   copyfile_flags_t flags = COPYFILE_ALL | COPYFILE_RECURSIVE;
   result = copyfile(fromPathStr, toPathStr, nullptr, flags);
They don't reply on the syntax of the path. There is a disparity here between Linux and Apple.

Reply:
I think the scope to enhance isDir() and recursive copying is much bigger than this ticket itself. I suggested that it should be done in a separated ticket (post helium).

Reply:
okay

Reply:
What happened to the initial change that Pasin suggested here?  That would do what these lines of code are attempting to replicate.